xgcm/test/test_grid_ufunc.py

```diff
@@ -81,6 +83,112 @@ def test_invalid_signatures(self, signature):
             _parse_grid_ufunc_signature(signature)
 
 
+class TestGetSignatureFromTypeHints:
+    def test_no_args_to_annotate(self):
+        @as_grid_ufunc()
+        def ufunc():
+            ...
+
+        assert ufunc.signature == "()->()"
```

Comment:
I'm not really sure if I'm somehow hiding some bugs by doing this. In normal numpy ufuncs doesn't `()` mean a scalar? Whereas here I'm using it to mean "no argument". I don't really test these cases but there might be bugs lurking here.

Reply:
I think we decided that this case is irrelevant because no conceivable grid ufunc would have zero inputs or zero outputs.

Reply:
I've now explicitly forbidden creating signatures which don't have either any input arguments or any output arguments (although I'm still not 100% in my head that there are no cases where you would want that). I think now there is a little inconsistency/inefficiency because the overall signature regex allows for zero or more arguments, when it should now allow for one or more.